win.go

```diff
@@ -74,8 +75,8 @@ func UTF16PtrToString(s *uint16) string {
 	return windows.UTF16PtrToString(s)
 }
 
-func MAKEINTRESOURCE(id uintptr) *uint16 {
-	return (*uint16)(unsafe.Pointer(id))
+func MAKEINTRESOURCE[ID constraints.Integer](id ID) *uint16 {
+	return (*uint16)(unsafe.Pointer(uintptr(id)))
```

Comment:
nit: the `constraints.Integer` constraint is too broad; a valid int resource ID is a `uint16`.
So maybe 
```suggestion
func MAKEINTRESOURCE[ID ~uint16](id ID) *uint16 {
	return (*uint16)(unsafe.Pointer(uintptr(id)))
```
or if there're reasons to accept signed integers and wider integers (e.g., `uint32` and `uint64`), then check that they are actually in the `uint16` range?

Reply:
My main concern there is that it's pretty common across the existing `walk` codebase to just pass literals, which in the absence of any other type, is just going to be an `int`. I think that is still desirable, so I'll throw in a panic check instead.